### DISCUSSION TITLE
Why is file name not complete when using `$@`

### DISCUSSION THREAD

**Discussion by nonetrix** [sxyazi/yazi#1145] [1. Q&A] [answered]
I have the following binding set, ignore the weird syntax it is compiled from Nix
```nix
[[manager.append_keymap]]
on = ["g", "r"]
run = "shell --confirm 'ripdrag $@"
```

But if a file has a space it seems to ignore everything after the space leading to
```
thread 'main' panicked at src/main.rs:111:9:
/home/noah/Downloads/Aprogamur : no such file or directory
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Exited with status code: 101
Process failed
```
I think this was working before a update, but could be wrong. I suspect this is user error though which is why I bring it up here instead of issues section, I could not figure this out myself. I think the file path should be escaped for it to work

**Comment by sxyazi** [comment, marked as answer]
It's missing the closing `'`. Also you should use `"$@"` instead of `$@` since the latter won't escape special characters in filenames, https://stackoverflow.com/questions/3898665/what-is-in-bash

**Comment by nonetrix** [comment]
Ah I think Nix might have done that nice catch

**Comment by nonetrix** [comment]
Yep that did it, feel dumb now. I had to do the following
```
run = "shell --confirm 'ripdrag \"$@\"'";
```
Those cursed escape sequences me want to throw my computer from the empire state building but it seems to work which is all that matters not touching it

**Reply by sxyazi** [reply]
Yeah this is a bit clunky, I'll [optimize it](https://github.com/sxyazi/yazi/issues/51#issuecomment-1913283446) in the next target. 

BTW, in TOML you can use `"""`, i.e. `run = """shell 'ripdrag "$@"' --confirm"""`, but I'm not sure if Nix has something similar.
